MIGRATION_GUIDE.md

```diff
@@ -1,14 +1,264 @@
+            let first_name = cx.argument::<JsString>(0)?;
+            let last_name = cx.argument::<JsString>(1)?;
+            Ok(User { first_name, last_name })
+        }
+
+        method full_name(mut cx) {
+            let this = cx.this();
+            let guard = cx.lock();
+            let user = this.borrow(&guard);
+            let full_name = user.full_name();
+            Ok(cx.string(full_name).upcast())
+        }
+    }
+}
+```
+
+**After:**
+
+```rust
+struct User {
+    first_name: String,
+    last_name: String,
+}
+
+fn create_user(mut cx: FunctionContext) -> JsResult<JsBox<User>> {
+    let first_name = cx.argument::<JsString>(0)?.value(&mut cx);
+    let last_name = cx.argument::<JsString>(1)?.value(&mut cx);
+    Ok(cx.boxed(User { first_name, last_name }))
+}
+
+fn user_full_name(mut cx: FunctionContext) -> JsResult<JsString> {
+    let user = cx.argument::<JsBox<User>>(0)?;
+    let full_name = user.full_name();
+    Ok(cx.string(full_name))
+}
+```
+
+### Concurrency
+
+The supported mechanism for concurrency is the Event Queue API (`neon::event::EventQueue`). This feature has not yet stabilized, so to use this API, you'll also need to enable the `"event-queue-api"` feature flag as well:
+
+```toml
+[dependencies.neon]
+version = "0.7.1"
+default-features = false
+features = ["napi", "event-queue-api"]
+```
+
+#### Deprecated: Task API
+
+The Task API (`neon::task`) is deprecated, and should in most cases be translated to using the Event Queue API.
+
+_Rationale:_ The Task API was built on top of the low-level libuv thread pool, which manages the concurrency of the Node.js system internals and should rarely be exposed to user-level programs. For most use cases, Neon users took advantage of this API as the only way to implement background, asynchronous computations. The Event Queue API is a more general-purpose, convenient, and safe way of achieving that purpose.
+
+That said, **if you believe you need access to the libuv thread pool, please [file an issue in the Neon repository](https://github.com/neon-bindings/neon/issues) with a description of your use case to let us know about it.** We don't believe this is commonly needed, but we don't want to leave you stuck! 
+
+**Before:**
+
+```rust
+impl Task for MyTask {
+    type Output = i32;
+    type Error = String;
+    type JsEvent = JsNumber;
+
+    fn perform(&self) -> Result<Self::Output, Self::Error> {
+        // compute the result...
+    }
+
+    fn complete(self, mut cx: TaskContext, result: Result<Self::Output, Self::Error>) -> JsResult<Self::JsEvent> {
+        match result {
+            Ok(n) => {
+                Ok(cx.number(n))
+            }
+            Err(s) => {
+                cx.throw_error(s)
+            }
+        }
+    }
+}
+
+pub fn start_task(mut cx: FunctionContext) -> JsResult<JsUndefined> {
+    let callback = cx.argument::<JsFunction>(0)?;
+    MyTask.schedule(callback);
+    Ok(cx.undefined())
+}
+```
+
+**After:**
```

Comment:
We might want to include the previous code would execute on the `libuv` thread pool, while this approach requires bringing your own thread pool.

Reply:
Great point, I added some more language around this -- see what you think.